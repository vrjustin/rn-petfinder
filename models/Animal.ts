interface Animal {
  id: number;
  organization_id: string;
  url: string;
  age: string;
  gender: string;
  size: string;
  coat: string;
  name: string;
  description: string;
  status: string;
  breeds: {
    mixed: boolean;
    primary: string;
    secondary: string | null;
    unknown: boolean;
  };
  colors: {
    primary: string;
    secondary: string | null;
    tertiary: string | null;
  };
  contact: {
    address: {
      address1: string;
      address2: string | null;
      city: string;
      country: string;
      postcode: string;
      state: string;
    };
    email: string;
    phone: string;
  };
  photos: {
    full: string;
    large: string;
    medium: string;
    small: string;
  }[];
  primary_photo_cropped: {
    full: string;
    large: string;
    medium: string;
    small: string;
  };
  species: string;
  tags: string[];
}

export default Animal;
