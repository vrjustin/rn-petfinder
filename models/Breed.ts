interface Breed {
  name: string;
  _links: {
    type: {
      href: string;
    };
  };
}

export default Breed;
