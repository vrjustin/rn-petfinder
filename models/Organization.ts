export interface Organization {
  id: string;
  name: string;
  email: string;
  phone: string;
  address: {
    address1: string;
    address2: string;
    city: string;
    state: string;
    postcode: string;
    country: string;
  };
  hours: {
    monday: string | null;
    tuesday: string | null;
    wednesday: string | null;
    thursday: string | null;
    friday: string | null;
    saturday: string | null;
    sunday: string | null;
  };
  url: string;
  website: string | null;
  mission_statement: string | null;
  adoption: {
    policy: string | null;
    url: string | null;
  };
  social_media: {
    facebook: string | null;
    twitter: string | null;
    youtube: string | null;
    instagram: string | null;
    pinterest: string | null;
  };
  photos: [
    {
      small: string;
      medium: string;
      large: string;
      full: string;
    },
  ];
  distance: number;
  _links: {
    self: {
      href: string;
    };
    animals: {
      href: string;
    };
  };
}

export default Organization;
