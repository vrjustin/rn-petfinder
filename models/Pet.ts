type PetType = 'dog' | 'cat' | 'bird';

interface Location {
    zipcode: string;
    distance: number;
}
function Pet() {

}

export default Pet;
