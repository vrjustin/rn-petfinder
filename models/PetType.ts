import {petTypesLocaleResolver} from '../strings/en.json';

const dogImage = require('../resources/black-1869685_1280.jpg');
const catImage = require('../resources/cat.jpg');
const rabbitImage = require('../resources/rabbit.jpg');
const smallFurryImage = require('../resources/chinchilla.jpg');
const horseImage = require('../resources/horse.jpg');
const birdImage = require('../resources/bird.jpg');
const lizardImage = require('../resources/lizard.jpg');
const llamaImage = require('../resources/lama.jpg');

export interface PetType {
  displayName: string;
  name: string;
  coats: string[];
  colors: string[];
  genders: string[];
  _links: {
    self: {href: string};
    breeds: {href: string};
  };
}

export const petTypeImages: Record<string, string> = {
  Dog: dogImage,
  Cat: catImage,
  Rabbit: rabbitImage,
  'Small & Furry': smallFurryImage,
  Horse: horseImage,
  Bird: birdImage,
  'Scales, Fins & Other': lizardImage,
  Barnyard: llamaImage,
};

export const mapTypeNameToLocaleName = (name: string): string | undefined => {
  switch (name) {
    case 'Dog':
      return petTypesLocaleResolver.Dog;
    case 'Cat':
      return petTypesLocaleResolver.Cat;
    case 'Rabbit':
      return petTypesLocaleResolver.Rabbit;
    case 'Small & Furry':
      return petTypesLocaleResolver.SmallFurry;
    case 'Horse':
      return petTypesLocaleResolver.Horse;
    case 'Bird':
      return petTypesLocaleResolver.Bird;
    case 'Scales, Fins & Other':
      return petTypesLocaleResolver.ScalesFinsOther;
    case 'Barnyard':
      return petTypesLocaleResolver.Barnyard;
  }
};
