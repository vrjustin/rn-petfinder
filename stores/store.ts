import {combineReducers, configureStore} from '@reduxjs/toolkit';
import petTypesReducer from '../reducers/petTypesReducer';
import petBreedsReducer from '../reducers/petBreedsReducer';
import animalsReducer from '../reducers/animalsReducer';
import searchParamsReducer from '../reducers/searchParamsReducer';

const createDebugger = require('redux-flipper').default;

const rootReducer = combineReducers({
  petTypes: petTypesReducer,
  petBreeds: petBreedsReducer,
  animals: animalsReducer,
  searchParameters: searchParamsReducer,
});

export type RootState = ReturnType<typeof rootReducer>;

const store = configureStore({
  reducer: rootReducer,
  middleware: getDefaultMiddleware =>
    __DEV__
      ? getDefaultMiddleware({serializableCheck: false}).concat(
          createDebugger(),
        )
      : getDefaultMiddleware({serializableCheck: false}),
});

export default store;
