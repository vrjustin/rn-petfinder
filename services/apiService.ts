import axios, {AxiosResponse, AxiosError} from 'axios';
import PetType from '../models/PetType';
import Breed from '../models/Breed';
import Animal from '../models/Animal';

interface AccessTokenResponse {
  access_token: string;
  token_type: string;
  expires_in: number;
}

interface PetTypesResponse {
  types: PetType[];
}

let jwt_access_token: string | null = null;

const AUTH_ENDPOINT = 'https://api.petfinder.com/v2/oauth2/token';
const TYPES_ENDPOINT = 'https://api.petfinder.com/v2/types';

const getAccessToken = async (): Promise<string | null> => {
  try {
    const response: AxiosResponse<AccessTokenResponse> = await axios.post(
      AUTH_ENDPOINT,
      {
        grant_type: 'client_credentials',
        client_id: process.env.CLIENT_ID!,
        client_secret: process.env.CLIENT_SECRET!,
      },
    );

    const {access_token} = response.data;
    jwt_access_token = access_token;
    return access_token;
  } catch (error) {
    console.log('-----------');
    console.log(error);
    console.log('-----------');
    console.error(error);
    return null;
  }
};

const getPetTypes = async (): Promise<PetTypesResponse> => {
  if (!jwt_access_token) {
    await getAccessToken();
  }

  try {
    const response: AxiosResponse<PetTypesResponse> = await axios.get(
      TYPES_ENDPOINT,
      {
        headers: {
          Authorization: `Bearer ${jwt_access_token}`,
        },
      },
    );

    return response.data;
  } catch (error) {
    console.error('Failed to get pet types:', error);
    return {types: []};
  }
};

const getPetBreeds = async (type: string): Promise<Breed[]> => {
  if (!jwt_access_token) {
    await getAccessToken();
  }
  try {
    const response = await axios.get(
      `https://api.petfinder.com/v2/types/${type}/breeds`,
      {
        headers: {
          Authorization: `Bearer ${jwt_access_token}`,
        },
      },
    );

    return response.data.breeds;
  } catch (error) {
    console.error('Failed to fetch pet breeds:', error);
    return [];
  }
};

const getAnimals = async (type: string, breed: string): Promise<Animal[]> => {
  if (!jwt_access_token) {
    await getAccessToken();
  }
  try {
    const response = await axios.get(
      `https://api.petfinder.com/v2/animals?type=${type}&breed=${breed}`,
      {
        headers: {
          Authorization: `Bearer ${jwt_access_token}`,
        },
      },
    );

    return response.data.animals;
  } catch (error: unknown) {
    console.error('Failed to fetch animals from type & breed:', error);
    if (axios.isAxiosError(error)) {
      const axiosError = error as AxiosError;
      if (axiosError.response?.status === 401) {
        console.log('Received 401 error, refreshing token...');
        await getAccessToken();
        return getAnimals(type, breed);
      }
    }
    console.error('Failed to fetch animals from type & breed: ', error);
    return [];
  }
};

export default {
  getAccessToken,
  getPetTypes,
  getPetBreeds,
  getAnimals,
};
