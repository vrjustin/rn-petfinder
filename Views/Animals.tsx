import React, {useState, useEffect} from 'react';
import {Text, View, FlatList, StyleSheet, TouchableOpacity, ScrollView} from 'react-native';
import {RouteProp} from '@react-navigation/native';
import {useNavigation} from '@react-navigation/native';
import Animal from '../models/Animal';
import Breed from '../models/Breed';
import apiService from '../services/apiService';
import PetType from '../models/PetType';
import FontAwesomeIcon from 'react-native-vector-icons/FontAwesome';

type RootStackParamList = {
  PetTypes: undefined;
  Breeds: {petType: PetType};
  Animals: {petType: string; selectedBreed: Breed};
};

type AnimalsScreenRouteProp = RouteProp<RootStackParamList, 'Animals'>;

type Props = {
  route: AnimalsScreenRouteProp;
};

const GridView: React.FC<{data: Animal[]}> = ({data}) => (
  <View style={styles.gridContainer}>
    {data.map(animal => (
      <TouchableOpacity
        key={animal.id}
        onPress={() => console.log('Pressed on an animal')}
        style={styles.gridItem}>
        <Text style={styles.text}>
          {animal.name} :: {animal.id}
        </Text>
      </TouchableOpacity>
    ))}
  </View>
);

const Animals: React.FC<Props> = ({route}) => {
  const [animals, setAnimals] = useState<Animal[]>([]);
  const [isGridView, setIsGridView] = useState(false);
  const {petType, selectedBreed} = route.params;
  const navigation = useNavigation();

  const toggleGridView = () => {
    setIsGridView(prevState => !prevState);
  };

  useEffect(() => {
    const fetchAnimalsData = async () => {
      try {
        const animalsData = await apiService.getAnimals(
          petType,
          selectedBreed.name,
        );
        setAnimals(animalsData);
      } catch (error) {
        console.error('Failed top fetch Breeds data: ', error);
      }
    };
    fetchAnimalsData();
  }, [petType, selectedBreed]);

  const handleAnimalSelection = (animal: Animal) => {
    console.log('Navigate to Animal Details : passed in is: ', animal);
    navigation.navigate('AnimalDetails', {selectedAnimal: animal});
  };

  const renderItem = ({item}: {item: Animal}) => (
    <TouchableOpacity onPress={() => handleAnimalSelection(item)}>
      <View style={styles.item}>
        <Text style={styles.text}>
          {item.name} :: {item.id}
        </Text>
      </View>
    </TouchableOpacity>
  );

  return (
    <View style={styles.container}>
      <View style={styles.headerRow}>
        <Text style={styles.header}>{selectedBreed.name} Pets</Text>
        <View style={styles.headerIconContainer}>
          <TouchableOpacity onPress={toggleGridView}>
            <FontAwesomeIcon
              name="th"
              size={20}
              color={isGridView ? 'black' : 'gray'}
              style={styles.icon}
            />
          </TouchableOpacity>
          <TouchableOpacity onPress={toggleGridView}>
            <FontAwesomeIcon
              name="list"
              size={20}
              color={!isGridView ? 'black' : 'gray'}
              style={styles.icon}
            />
          </TouchableOpacity>
        </View>
      </View>
      {isGridView ? (
        <ScrollView>
          <GridView data={animals} />
        </ScrollView>
      ) : (
        <FlatList
          data={animals}
          renderItem={renderItem}
          keyExtractor={item => item.id.toString()}
        />
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 20,
    backgroundColor: '#f0f0f0',
  },
  headerRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  header: {
    fontSize: 20,
    fontWeight: 'bold',
    marginBottom: 20,
  },
  headerIconContainer: {
    flexDirection: 'row',
  },
  flatListContent: {
    flexGrow: 1,
  },
  item: {
    backgroundColor: '#fff',
    padding: 15,
    marginBottom: 10,
    borderRadius: 5,
    elevation: 3,
  },
  text: {
    fontSize: 16,
  },
  icon: {
    marginRight: 8,
  },
  gridContainer: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'space-between', // Adjust as needed
    paddingHorizontal: 10, // Add some padding
  },
  gridItem: {
    width: '48%', // Two items per row with a small gap
    backgroundColor: '#fff',
    padding: 15,
    marginBottom: 10,
    borderRadius: 5,
    elevation: 3,
  },
});

export default Animals;
