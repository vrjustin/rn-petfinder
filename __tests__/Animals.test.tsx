import React from 'react';
import {Provider} from 'react-redux';
import store from '../stores/store';
import renderer, {act} from 'react-test-renderer';
import Animals from '../Views/Animals';
import {PetType} from '../models/PetType';
import Breed from '../models/Breed';
import {AnimalResults} from '../services/apiService';

jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({
    navigate: jest.fn(),
    setOptions: jest.fn(),
  }),
}));

jest.mock('@react-native-async-storage/async-storage', () => ({
  setItem: jest.fn(),
  getItem: jest.fn(),
}));

const animalResults: AnimalResults = {
  animalsData: [
    {
      id: 1,
      organization_id: 'org1',
      url: 'https://example.com',
      age: 'Young',
      gender: 'Female',
      size: 'Medium',
      coat: 'Short',
      name: 'Fluffy',
      description: 'A fluffy dog',
      status: 'Available',
      breeds: {
        mixed: false,
        primary: 'Poodle',
        secondary: null,
        unknown: false,
      },
      colors: {
        primary: 'White',
        secondary: null,
        tertiary: null,
      },
      contact: {
        address: {
          address1: '123 Main St',
          address2: null,
          city: 'Anytown',
          country: 'USA',
          postcode: '12345',
          state: 'NY',
        },
        email: 'email@example.com',
        phone: '555-123-4567',
      },
      photos: [
        {
          full: 'https://example.com/full.jpg',
          large: 'https://example.com/large.jpg',
          medium: 'https://example.com/medium.jpg',
          small: 'https://example.com/small.jpg',
        },
      ],
      primary_photo_cropped: {
        full: 'https://example.com/full.jpg',
        large: 'https://example.com/large.jpg',
        medium: 'https://example.com/medium.jpg',
        small: 'https://example.com/small.jpg',
      },
      species: 'Dog',
      tags: ['Friendly', 'Playful'],
      isFavorite: false,
    },
  ],
  pagination: {current_page: 1, total_pages: 1},
};

const mockedPetType: PetType = {
  displayName: 'Dog',
  name: 'Dog',
  coats: [],
  colors: [],
  genders: [],
  _links: {
    self: {href: ''},
    breeds: {href: ''},
  },
};

const mockBreeds: Breed[] = [
  {
    name: 'Poodle',
    _links: {
      type: {href: ''},
    },
  },
];

let renderedAnimalsTree: any;

jest.mock('../services/apiService.ts', () => ({
  getAnimals: jest.fn(() => animalResults),
}));

describe('Animals', () => {
  it('renders correctly', async () => {
    await act(async () => {
      renderedAnimalsTree = renderer.create(
        <Provider store={store}>
          <Animals
            route={{
              params: {
                petType: mockedPetType,
                selectedBreeds: mockBreeds,
              },
            }}
          />
        </Provider>,
      );
    });
    expect(renderedAnimalsTree.toJSON()).toMatchSnapshot();
  });
});
