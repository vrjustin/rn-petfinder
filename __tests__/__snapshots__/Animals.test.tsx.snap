// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animals renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
      "padding": 0,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": 20,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 20,
        }
      }
    >
      Poodle
       Pets
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "black",
                "fontSize": 20,
              },
              {
                "marginRight": 8,
              },
              {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "gray",
                "fontSize": 20,
              },
              {
                "marginRight": 8,
              },
              {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <RCTSafeAreaView
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "margin": 8,
        }
      }
    >
      <RCTScrollView
        contentContainerStyle={
          {
            "flexGrow": 1,
          }
        }
        data={
          [
            {
              "age": "Young",
              "breeds": {
                "mixed": false,
                "primary": "Poodle",
                "secondary": null,
                "unknown": false,
              },
              "coat": "Short",
              "colors": {
                "primary": "White",
                "secondary": null,
                "tertiary": null,
              },
              "contact": {
                "address": {
                  "address1": "123 Main St",
                  "address2": null,
                  "city": "Anytown",
                  "country": "USA",
                  "postcode": "12345",
                  "state": "NY",
                },
                "email": "email@example.com",
                "phone": "555-123-4567",
              },
              "description": "A fluffy dog",
              "gender": "Female",
              "id": 1,
              "isFavorite": false,
              "name": "Fluffy",
              "organization_id": "org1",
              "photos": [
                {
                  "full": "https://example.com/full.jpg",
                  "large": "https://example.com/large.jpg",
                  "medium": "https://example.com/medium.jpg",
                  "small": "https://example.com/small.jpg",
                },
              ],
              "primary_photo_cropped": {
                "full": "https://example.com/full.jpg",
                "large": "https://example.com/large.jpg",
                "medium": "https://example.com/medium.jpg",
                "small": "https://example.com/small.jpg",
              },
              "size": "Medium",
              "species": "Dog",
              "status": "Available",
              "tags": [
                "Friendly",
                "Playful",
              ],
              "url": "https://example.com",
            },
          ]
        }
        getItem={[Function]}
        getItemCount={[Function]}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={0.0001}
        stickyHeaderIndices={[]}
        viewabilityConfigCallbackPairs={[]}
      >
        <View>
          <View
            onFocusCapture={[Function]}
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    {
                      "aspectRatio": 1,
                      "flex": 1,
                      "margin": 4,
                      "width": 359,
                    }
                  }
                >
                  <View
                    accessibilityIgnoresInvertColors={true}
                    style={
                      {
                        "alignItems": "center",
                        "flex": 1,
                        "justifyContent": "center",
                        "resizeMode": "cover",
                      }
                    }
                  >
                    <Image
                      source={
                        {
                          "uri": "https://example.com/medium.jpg",
                        }
                      }
                      style={
                        [
                          {
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          },
                          {
                            "height": undefined,
                            "width": undefined,
                          },
                          {
                            "borderRadius": 8,
                          },
                        ]
                      }
                    />
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "borderRadius": 8,
                          "flex": 1,
                          "justifyContent": "center",
                          "margin": 4,
                          "padding": 16,
                        }
                      }
                    >
                      <View
                        style={
                          {
                            "position": "absolute",
                            "right": -68,
                            "top": 8,
                          }
                        }
                      >
                        <View
                          accessibilityState={
                            {
                              "busy": undefined,
                              "checked": undefined,
                              "disabled": undefined,
                              "expanded": undefined,
                              "selected": undefined,
                            }
                          }
                          accessibilityValue={
                            {
                              "max": undefined,
                              "min": undefined,
                              "now": undefined,
                              "text": undefined,
                            }
                          }
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            {
                              "opacity": 1,
                            }
                          }
                        >
                          <Text
                            allowFontScaling={false}
                            selectable={false}
                            style={
                              [
                                {
                                  "color": "white",
                                  "fontSize": 20,
                                },
                                {
                                  "marginRight": 8,
                                },
                                {
                                  "fontFamily": "FontAwesome",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                {},
                              ]
                            }
                          >
                            
                          </Text>
                        </View>
                      </View>
                      <View
                        style={
                          {
                            "bottom": 4,
                            "position": "absolute",
                          }
                        }
                      >
                        <View
                          style={
                            {
                              "flexDirection": "row",
                            }
                          }
                        >
                          <Text
                            style={
                              {
                                "color": "white",
                                "fontSize": 16,
                                "fontWeight": "bold",
                              }
                            }
                          >
                            Fluffy
                            ,
                          </Text>
                          <Text
                            style={
                              {
                                "color": "white",
                                "fontSize": 16,
                              }
                            }
                          >
                             
                            Young
                          </Text>
                        </View>
                        <View
                          style={
                            {
                              "flexDirection": "row",
                            }
                          }
                        >
                          <Text
                            allowFontScaling={false}
                            selectable={false}
                            style={
                              [
                                {
                                  "color": "white",
                                  "fontSize": 20,
                                },
                                {
                                  "marginRight": 8,
                                },
                                {
                                  "fontFamily": "FontAwesome",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                {},
                              ]
                            }
                          >
                            
                          </Text>
                          <Text
                            style={
                              {
                                "color": "white",
                                "fontSize": 16,
                              }
                            }
                          >
                            Anytown
                            ,
                            NY
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>
</View>
`;
